vip-cache-manager/js/admin-bar.js

```diff
@@ -0,0 +1,69 @@
+(async () => {
+	async function postData(url = '', data = {}) {
+		// Default options are marked with *
+		const response = await fetch(url, {
+			method: 'POST',
+			credentials: 'same-origin',
+			headers: {
+				'Content-Type': 'application/json'
+			},
+			body: JSON.stringify({ ...data })
+		});
+		return response.json();
+	}
+
+	// Indicates whether there's a purge request happening
+	let purgeQueried = false;
+	// Stores the ref to the DOM node
+	let btn;
+
+	/**
+	 * Grab all the necesary URLs (incl. scripts and CSS) for the purge.
+	 */
+	const getURLsToPurge = () => {
+		return [document.location.toString()].concat(
+			Array.from(document.querySelectorAll('script[src]')).map(({ src }) => src),
+			Array.from(document.querySelectorAll('link[rel=stylesheet]')).map(({ href }) => href)
```

Comment:
I was wondering if we should include images as well, but probably not necessary since they're not impacted by deploys, which is when this is most needed.

Reply:
I'm wondering whether that `filter` is needed here, maybe just leave the validation to the backend altogether?

Reply:
I would love to have images included. We've got Media customers who have had to fully delete an image due to legal reasons, and while they could delete the image from the media library, the image was cached at the VIP edge and it was a pain to try and clear it.

If we're not going to provide a free-form text field where URLs could be entered to purge (image) URLs one by one, then I think a "Flush page cache" button should flush _everything_ on that page, including scripts, styles, images, uploaded videos and PDFs and any other assets referenced on the page.

Reply:
@GaryJones An image delete from the media library should trigger a purge; was that not the case?

Reply:
I guess it may have been fixed now, but it was previously broken - deleted media library item didn't delete. I guess it could also have been down to https://github.com/Automattic/vip-go-mu-plugins/blob/5fd43e123c0cc0f2da925201fe2484bcf14b5712/vip-cache-manager/vip-cache-manager.php#L346-L349.

Let's skip over my concern if it's believed to be handled elsewhere then.